tensorflow/core/ops/dataset_ops.cc

```diff
@@ -399,6 +399,22 @@ batch_size: A scalar representing the number of elements to accumulate in a
   batch.
 )doc");
 
+REGISTER_OP("SlideDataset")
+    .Input("input_dataset: variant")
+    .Input("slide_size: int64")
+    .Input("slide_step: int64")
+    .Output("handle: variant")
+    .Attr("output_types: list(type) >= 1")
+    .Attr("output_shapes: list(shape) >= 1")
+    .SetShapeFn(shape_inference::ScalarShape)
+    .Doc(R"doc(
+Creates a dataset that slides a fixed window on `input_dataset`.
+
+slide_size: A scalar representing the number of elements in the fixed window.
+slide_step: A scalar representing the steps moving forward in one iteration.
+  The default is `1`. It must be in [1, `slide_size`).
```

Comment:
Note that there are no defaults at this level of the API (for tensor-valued inputs), so please delete this line.